include/ck_tile/ops/gemm/pipeline/block_gemm_pipeline_agmem_bgmem_creg_universal_policy.hpp

```diff
@@ -0,0 +1,421 @@
+
+        static_assert(kM1 != 0, "M1 is zero, which will lead to a division by zero error.");
+
+        return make_static_tile_distribution(
+            tile_distribution_encoding<sequence<1>,
+                                       tuple<sequence<kM0, kM1>, sequence<kK0, kK1>>,
+                                       tuple<sequence<1>, sequence<2>>,
+                                       tuple<sequence<0>, sequence<1>>,
+                                       sequence<1, 2>,
+                                       sequence<1, 0>>{});
+    }
+
+    template <typename Problem>
+    CK_TILE_HOST_DEVICE static constexpr auto MakeBDramTileDistribution()
+    {
+        using WarpGemm = WarpGemmMfmaDispatcher<typename Problem::ADataType,
+                                                typename Problem::BDataType,
+                                                typename Problem::CDataType,
+                                                Problem::BlockGemmShape::WarpTile::at(I0),
+                                                Problem::BlockGemmShape::WarpTile::at(I1),
+                                                Problem::BlockGemmShape::WarpTile::at(I2),
+                                                true>;
+
+        constexpr index_t kNPerBlock = Problem::BlockGemmShape::kN;
+        constexpr index_t kKPerBlock = Problem::BlockGemmShape::kK;
+
+        constexpr auto kN0 = get_warp_size() * Problem::BlockGemmShape::BlockWarps::at(I1);
+        constexpr auto kN1 = kNPerBlock / kN0;
+
+        constexpr index_t kK1 = WarpGemm::kK;
+        constexpr index_t kK0 = kKPerBlock / kK1;
+
+        static_assert(kN1 != 0, "M1 is zero, which will lead to a division by zero error.");
+
+        return make_static_tile_distribution(
+            tile_distribution_encoding<sequence<1>,
+                                       tuple<sequence<kN0, kN1>, sequence<kK0, kK1>>,
+                                       tuple<sequence<1>, sequence<2>>,
+                                       tuple<sequence<0>, sequence<1>>,
+                                       sequence<1, 2>,
+                                       sequence<1, 0>>{});
+    }
+
+    template <typename Problem>
+    CK_TILE_HOST_DEVICE static constexpr auto GetBlockGemm()
+    {
+        using AccDataType     = float;
+        using BlockWarps      = typename Problem::BlockGemmShape::BlockWarps;
+        using WarpTile        = typename Problem::BlockGemmShape::WarpTile;
+        using WarpGemm        = WarpGemmMfmaDispatcher<typename Problem::ADataType,
+                                                typename Problem::BDataType,
+                                                AccDataType,
+                                                WarpTile::at(I0),
+                                                WarpTile::at(I1),
+                                                WarpTile::at(I2),
+                                                true>;
```

Comment:
Is that indentation correct? Could we shift right this to align with first tparam ?